### PR TITLE
Added utility for resolving gas prices

### PR DESCRIPTION
**Pull request by nlordell**
Added a utility for resolving gas prices. This is currently not used anywhere in the code base, but will be used by #137 and was split out to make review a bit easier.

### Test Plan

CI with new test cases.

### CODE REVIEW

File: src/transaction/gas_price.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    /// transactions, or the transactions not being mined at all.
+    pub fn low() -> Self {
+        GasPrice::Scaled(0.8)
+    }
+
+    /// A high gas price that usually results in faster mining times.
+    /// transactions, or the transactions not being mined at all.
+    pub fn fast() -> Self {
+        GasPrice::Scaled(6.0)
+    }
```
Comment (bh2smith) [inline src/transaction/gas_price.rs]: I see we have "low" and "fast". Is this a clash of naming?

**Review by bh2smith — APPROVED** [review]
This is very nice! Just a question about the clashing names.

File: src/transaction/gas_price.rs (hunk 1 above)
Comment (nlordell) [reply src/transaction/gas_price.rs]: `fast` was changed to `high`

File: src/transaction/gas_price.rs [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    /// gas price will be scaled once retrieved.
+    Estimating(#[pin] CompatCallFuture<T, U256>, Option<f64>),
+}
+
+impl<T: Transport> ResolveGasPriceFuture<T> {
+    /// Creates a new future that resolves once the gas price value is computed.
+    pub fn new(web3: &Web3<T>, gas_price: GasPrice) -> Self {
+        let state = match gas_price {
+            GasPrice::Standard => {
+                ResolveGasPriceState::Estimating(web3.eth().gas_price().compat(), None)
```
Comment (fleupold) [inline src/transaction/gas_price.rs:135]: Nit, could be 1.0?

File: src/transaction/gas_price.rs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            GasPrice::Value(value) => Some(*value),
+            _ => None,
+        }
+    }
+
+    /// Resolves the gas price into a value. Returns a future that resolves once
+    /// the gas price is calculated as this may require contacting the node for
+    /// gas price estimates in the case of `GasPrice::Standard` and
+    /// `GasPrice::Scaled`.
+    pub fn resolve<T: Transport>(web3: &Web3<T>, gas_price: Self) -> ResolveGasPriceFuture<T> {
```
Comment (fleupold) [inline src/transaction/gas_price.rs]: This method makes testing easier? Or why don't we want to inline it?

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2Fxqnq1v2eqioOQ%2Fgiphy.gif&token=gas_price" data-gitmeme-token="gas_price"></a>

File: src/transaction/gas_price.rs (hunk 2 above)
Comment (nlordell) [reply src/transaction/gas_price.rs:135]: Then we could have rounding issues.

File: src/transaction/gas_price.rs (hunk 3 above)
Comment (nlordell) [reply src/transaction/gas_price.rs]: Do you mean why not make it take `&self`? For consistency with `GasPrice::resolve_for_transaction_request`. Not really attached to it being an associated (static) function and if you find it easier to read if it took a `self` parameter I can change it.

File: src/transaction/gas_price.rs (hunk 2 above)
Comment (nlordell) [reply src/transaction/gas_price.rs:135]: Tried to improve the comments as to why `GasPrice::Standard` and `GasPrice::Scaled(1.0)` are slightly different:
https://github.com/gnosis/ethcontract-rs/blob/5843672cbdf2ff2b1c54aa6c0051ab4173ccfb37/src/transaction/gas_price.rs#L23-L27

File: src/transaction/gas_price.rs (hunk 3 above)
Comment (nlordell) [reply src/transaction/gas_price.rs]: The other motivating factor was that conceptually, I felt like `Web3::resolve_gas_price(GasPrice)` makes more to me, in that it should be a method on `Web3` and not `GasPrice`.

Comment (nlordell) [reply src/transaction/gas_price.rs]: I changed it :)

**Pull request merged**